projects/vdk-plugins/vdk-impala/src/vdk/plugin/impala/impala_error_handler.py

```diff
@@ -107,6 +109,7 @@ def _handle_exception(self, exception, recovery_cursor) -> bool:
             or self._handle_metadata_exception_with_invalidate_and_retry(
                 exception, recovery_cursor
             )
+            or self._handle_memory_error(exception, recovery_cursor)
```

Comment:
There are better patterns for this kind of error handling. You can do some kind of handler chain here. Doesn't have to be in this PR, but it wouldn't hurt to open a refactoring ticket.

Reply:
I would like to focus this PR on this specific issue and honestly I won't have the time to refactor this code or I don't think anyone else would "just cause"...